### DISCUSSION TITLE
Moving source code off GitHub

### DISCUSSION THREAD

**Discussion by vhscom** [FreeTubeApp/FreeTube#2047] [Q&A] [answered]
I just read the [October 2021 blog post](https://blog.freetubeapp.io/behind-the-scenes-improving-project-structure-and-other-updates) and the first thing I thought when I read maintenance was becomming a burden was why are they hosting on GitHub. There are very good open and non-commercial alternatives to GitHub such as [Codeberg.org](https://codeberg.org/) who I'm sure could benefit greatly from the exposure of a project like this utilizing them and their new CI offering. Why not take advantage?

**Comment by PrestonN** [comment, marked as answer]
First off, I personally do really like Codeberg and I appreciate the work that they do. I also have a private Gitea instance that I use for personal projects that I don't have public as well as for mirroring several repositories (FreeTube itself included). While I really like a lot about Codeberg / Gitea, I don't really see us moving over to it. There's a couple of reasons why it's best to stay in GitHub:

- In reference to the blog post I made, if maintenance and lack of contributors is the issue, then it's unlikely that moving to a smaller platform will help with this. Being on GitHub, at least in some way, adds _a lot_ of visibility to projects. GitHub is a very large platform and we benefit a lot from it. Even large projects like GNOME have a read only mirror of their code on GitHub specifically to attract more developers.
- As far as privacy is concerned, everything we do is public regardless of where we're at, so there isn't really any privacy loss by staying here versus another platform. Those that are concerned with GitHub / Microsoft are welcome to use our fairly active Matrix community to report issues if that's what they'd prefer.
- We use GitHub Actions _a lot_. Even if the Codeberg CI had everything that we'd need, I would feel more guilty for using it because it's a smaller service with much smaller infrastructure. Using their CI has a very real cost that they have to make and misuse of it could be pretty taxing to what they have. With as large as GitHub is, the same concern just isn't there.
- Back to CI, we have a fairly complicated build process. It's kinda difficult finding information about how Codeberg has their CI setup, but I'm not sure if they have support for non-Linux runners. We need something that can create binaries for Windows / Mac / Linux and it's fairly difficult finding something that has all 3. With as small as Codeberg is, they likely don't have Mac systems to use, which would prevent us from creating builds for that platform.
- Along with building for the main 3 operating systems, we also publish FreeTube onto Flathub. While Flathub is for distributing Linux apps,  they have their own process for publishing. We use GitHub Actions to automate this entire process more or less. Because Flathub also uses GitHub for this process, it only makes the process more complicated and difficult if we were somewhere else.

Moving from GitHub is a good idea at a glance and if our scope was a bit smaller, it might be feasible, though it's much easier on myself and the team if we stay on the platform. I will definitely be considering Codeberg / Gitea if I ever create a smaller project, but for FreeTube, I unfortunately don't see it happening.

**Reply by vhscom** [reply]
Thanks for the thoughtful responses, Preston. And I'm thrilled to hear the exposure GitHub is providing isn't the burden I'd thought it might be. Certainly moving to [Woodpecker CI](https://woodpecker-ci.org/) would come with its challenges. When I think of apps like NewPipe and FreeTube I worry that at some point GitHub will censor them. Let's just hope it doesn't come to that. Cheers my friend. :smile:
